include/qpdf/QPDFTokenizer.hh

```diff
@@ -219,7 +220,17 @@ class QPDFTokenizer
     bool nextToken(
         InputSource& input, std::string const& context, size_t max_len = 0);
 
-  private:
+    // The following methods are only valid after nextToken has been called
+    // and until another QPDFTokenizer method is called. They allow the results
+    // of calling nextToken to be accessed without creating a Token, thus
+    // avoiding copying information that may not be needed.
+    // NB Any string_views returned by these methods will be invalid after
```

Comment:
Maybe we should never return std::string_view at all.

Reply:
I added some notes about this to #785.

Reply:
I think this might be a case where returning std::string_view would be okay since this is a very specialized interface and performance critical, but if we do that, we should do it now and not start with `std::string const&` and change it later.

Reply:
And your NB comment provides the needed warning.

Reply:
> Are any string_views returned?

Not any longer. Comment removed.

> Maybe we should never return std::string_view at all.

Agreed for the public API.

For the private API it there may be a case, but it should limited to
- where it is performance critical, and
- where the string_view is immediately consumed.

I suspect that there probably is no need to return string_views in qpdf across class boundaries.

Both for the public and private API there are probably cases where we should return a string&& instead of a string const&.

Reply:
I never considered returning a `string&&`. I guess that lets you return a temporary string without copying it? And I suppose auto would still be string. I guess since this is a public API, we can leave it as `std::string const&` for now.